extensions-contrib/materialized-view-maintenance/src/test/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisorTest.java

```diff
@@ -176,6 +176,29 @@ public void testCheckSegments() throws IOException
     Assert.assertEquals(expectedSegments, toBuildInterval.rhs);
   }
 
+  @Test
+  public void testCheckSegmentsAndSubmitTasks() throws IOException
```

Comment:
I don't think this test checks your change because there's no running tasks. I think the easiest way is to add a method annotated by `@VisibleForTesting` which updates `runningTasks` and calls `checkSegmentsAndSubmitTasks()`.